sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/CertificateAsyncClient.java

```diff
@@ -846,17 +841,43 @@ public String getVaultUrl() {
     public PagedFlux<DeletedCertificate> listDeletedCertificates() {
         try {
             return new PagedFlux<>(
-                () -> withContext(context -> listDeletedCertificatesFirstPage(context)),
+                () -> withContext(context -> listDeletedCertificatesFirstPage(false, context)),
                 continuationToken -> withContext(
                     context -> listDeletedCertificatesNextPage(continuationToken, context)));
         } catch (RuntimeException ex) {
             return new PagedFlux<>(() -> monoError(logger, ex));
         }
     }
 
-    PagedFlux<DeletedCertificate> listDeletedCertificates(Context context) {
+    /**
+     * Lists the {@link DeletedCertificate deleted certificates} in the key vault currently available for recovery. This
+     * operation includes deletion-specific information and is applicable for vaults enabled for soft-delete. This
+     * operation requires the {@code certificates/get/list} permission.
+     *
+     * <p><strong>Code Samples</strong></p>
+     * <p>Lists the deleted certificates in the key vault. Prints out the
+     * recovery id of each deleted certificate when a response has been received.</p>
+     *
+     * {@codesnippet com.azure.security.keyvault.certificates.CertificateAsyncClient.listDeletedCertificates}
+     *
+     * @param includePending indicate if pending deleted certificates should be included in the results.
+     * @return A {@link PagedFlux} containing all of the {@link DeletedCertificate deleted certificates} in the vault.
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedFlux<DeletedCertificate> listDeletedCertificates(Boolean includePending) {
+        try {
+            return new PagedFlux<>(
+                () -> withContext(context -> listDeletedCertificatesFirstPage(includePending, context)),
+                continuationToken -> withContext(
+                    context -> listDeletedCertificatesNextPage(continuationToken, context)));
+        } catch (RuntimeException ex) {
+            return new PagedFlux<>(() -> monoError(logger, ex));
+        }
+    }
+
+    PagedFlux<DeletedCertificate> listDeletedCertificates(Boolean includePending, Context context) {
         return new PagedFlux<>(
-            () -> listDeletedCertificatesFirstPage(context),
+            () -> listDeletedCertificatesFirstPage(includePending, context),
             continuationToken -> listDeletedCertificatesNextPage(continuationToken, context));
```

Comment:
listDeletedCertificatesNextPage doesn't need the parameter `includePending`?

Reply:
nope,
next page just needs the continuationToken.